Flow.Launcher/PublicAPIInstance.cs

```diff
@@ -33,19 +34,26 @@ public class PublicAPIInstance : IPublicAPI
     {
         private readonly SettingWindowViewModel _settingsVM;
         private readonly MainViewModel _mainVM;
-        private readonly PinyinAlphabet _alphabet;
 
-        #region Constructor
+        private Updater _updater;
 
-        public PublicAPIInstance(SettingWindowViewModel settingsVM, MainViewModel mainVM, PinyinAlphabet alphabet)
+        #region Constructor & Initialization
+
+        public PublicAPIInstance()
         {
-            _settingsVM = settingsVM;
-            _mainVM = mainVM;
-            _alphabet = alphabet;
+            _settingsVM = Ioc.Default.GetRequiredService<SettingWindowViewModel>();
+            _mainVM = Ioc.Default.GetRequiredService<MainViewModel>();
             GlobalHotkey.hookedKeyboardCallback = KListener_hookedKeyboardCallback;
             WebRequest.RegisterPrefix("data", new DataWebRequestFactory());
         }
 
+        public void Initialize()
+        {
+            // We need to initialize Updater not in the constructor because we want to avoid
+            // recrusive dependency injection
+            _updater = Ioc.Default.GetRequiredService<Updater>();
```

Comment:
I remember this is one of the reason I forfeit DI before. Not elegant but maybe good enough.